server-src/src/main/java/com/itt/kmt/services/ArticleService.java

```diff
@@ -0,0 +1,140 @@
+
+import java.util.List;
+
+
+/**
+ * Service class that acts as an intermediary between controller and the
+ * database for all basic CRUD operations. The business logic should reside in
+ * service class.
+ * 
+ * @author Ashish Y
+ */
+@Service
+public class ArticleService {
+
+    /**
+     * Instance of the basic Repository implementation.
+     */
+    @Autowired
+    private ArticleRepository articleRepository;
+
+    /**
+     * Instance of user service.
+     */
+    @Autowired
+    private UserService userService;
+
+    /**
+     * Instance of the basic Article Type repository.
+     */
+    @Autowired
+    private ArticleTypeRepository articleTypeRepository;
+
+    /**
+     * Saves the Article to database.
+     * 
+     * @param article Article object to be saved
+     * @return Article object with userResponse included.
+     */
+    public Article save(final Article article) {
+
+        if (article.getCreatedBy() != null) {
+            User createdByUser = userService.getUserByID(article.getCreatedBy().toString());
+            article.setCreatedBy(convertUserIntoUserResponse(createdByUser));
+        }
+        if (article.getApprover() != null) {
+            User approver = userService.getUserByID(article.getApprover().toString());
+            article.setApprover(convertUserIntoUserResponse(approver));
+        }
+        if (article.getArticleType() != null) {
+            article.setArticleType(getArticleTypeByID(article.getArticleType().toString()));
+        }
+
+        return articleRepository.save(article);
+    }
+
+    /**
+     * Convert user object into article response user format.
+     *
+     * @param user User object to be converted
+     * @return article user response details.
+     */
+    private UserResponse convertUserIntoUserResponse(final User user) {
+        UserResponse userResponse = new UserResponse(user.getId(),
+                user.getFirstName(), user.getLastName(), user.getEmail());
+        return userResponse;
+    }
+
+    /**
+     * Get all the Article Types.
+     * @return List of all the Article Type.
+     */
+    public List<ArticleType> getArticleTypes() {
+        return (List<ArticleType>) articleTypeRepository.findAll();
+    }
+
+    /**
+     * Get Article Type with ID.
+     * @param id Article ID.
+     * @return Article Type.
+     */
+    private ArticleType getArticleTypeByID(final String id) {
+        return articleTypeRepository.findOne(id);
+    }
+
+    /**
+     * updates the DBEntity(Article) from the database.
+     * 
+     * @param id of Article to be updated.
+     * @param updatedArticle , Article object that needs to be updated.
+     * @return Article
+     */
+    public Article updateArticle(final String id, final Article updatedArticle) {
+        Article article = articleRepository.findOne(id);
+        if (article == null) {
+            throw new RuntimeException("Article not found");
+        }
+        article.setTitle(updatedArticle.getTitle());

```

Comment:
To be on safer side, check for emptyness, if not, then only set the value
if ( ! updateArticle.getTitle().isEmpty() ) {
    // set title
}

Applicable to other setters also